apps/comments-ui/src/actions.ts

```diff
@@ -186,9 +186,29 @@ async function showComment({state, api, data: comment}: {state: EditableAppConte
     };
 }
 
-async function likeComment({state, api, data: comment}: {state: EditableAppContext, api: GhostApi, data: {id: string}}) {
-    await api.comments.like({comment});
+async function sendLikeToApi({api, data: comment}: {api: GhostApi, data: {id: string, commentsState: Comment[]}}) {
+    const commentsState = comment.commentsState;
 
+    try {
+        await api.comments.like({comment: {id: comment.id}});
+        return {
+            commentLikeLoadingId: null
+        };
+    } catch (err) {
+        // if error we revert the state
+        return {
+            comments: commentsState,
+            commentLikeLoadingId: null
+        };
```

Comment:
This seems dangerous as other state changes could have happened if there was a long delay before the endpoint eventually errored. I would expect a rollback to only occur for the liked status of the comment under change, would that make things considerably more difficult?